src/pymorize/units.py

```diff
@@ -0,0 +1,108 @@
+import pint
+import re
+
+
+"""
+This module deals with the auto-unit conversion in the cmorization process.
+In case the units in model files differ from CMIP Tables, this module attempts to convert them automatically.
+
+In case of missing units in either model files or CMIP Tables, this module can not convert from a dimentionless base to something with dimension. Dealing with such thing have to done with `action` section in the Rules module on a per variable basis.
+
+"""
+
+
+# TODO: decide a place to hold the missing unit definitions that pint is not aware of
+# from pathlib import Path
+# _unitsfile = Path(__file__).parent / "units_en.txt"
+# u = pint.UnitRegistry(_unitsfile)
+
+
+ureg = pint.UnitRegistry()
+# ureg.define('degC = degree_Celsius')
+# https://ncics.org/portfolio/other-resources/udunits2/
+ureg.define("degrees_east = deg")
+ureg.define("degree_east = deg")
+ureg.define("degrees_north = deg")
+ureg.define("degree_north = deg")
+ureg.define("degrees_west = -1 * deg")
+ureg.define("degrees_south = -1 * deg")
+# https://planetcalc.com/6777/
+ureg.define("molN = 14.007 * g")
+ureg.define("molC = 12.0107 * g")
+ureg.define("molFe = 55.874 * g")
+
+
+def format_exponent_notation(s, pattern=re.compile(r"(?P<name>\w+)-(?P<exp>\d+)")):
+    "m-2 -> m^-2"
+
+    def correction(match):
+        try:
+            float(match.group())
+        except ValueError:
+            d = match.groupdict()
+            s = f"{d['name']}^-{d['exp']}"
+            return s
+        return match.group()
+
+    return re.sub(pattern, correction, s)
+
+
+def format_power_notation(s, pattern=re.compile(r"(?P<name>\w+)(?P<exp>\d+)")):
+    "m2 -> m^2"
+
+    def correction(match):
+        try:
+            float(match.group())
+        except ValueError:
+            d = match.groupdict()
+            s = "{0[name]}^{0[exp]}".format(d)
+            if d["exp"] == 1:
+                s = "{0[name]}".format(d)
+            return s
+        return match.group()
+
+    return re.sub(pattern, correction, s)
```

Comment:
For both of these functions, there can be some improvements:
* better function name can be used to improve clarity
* More explicit docstrings

Reply:
In my first attempt, the name I came up with is `fix_exponent_notation` and `fix_power_notation` as the purpose of these functions is to correct the short comings in unit notation so pint can read it. It was suggested to me to change the names to `format_exponent_notation` and `format_power_notation` to improve clarity.  Alternative names: normalize_exponent_notation, standardize_exponent_notation. I think this function should start with a leading underscore as it is just a implementation detail and will never a part of API.

I will elaborate on documentation string.